Source/Eto/Forms/FilterCollection.cs

```diff
@@ -785,11 +785,9 @@ public virtual void Insert(int index, T item)
 					bool directInsert = sort == null && matchesFilter;
 					if (index > 0)
 					{
-						var beforeItem = filtered[index - 1];
-						var itemsIndex = items.IndexOf(beforeItem);
-						items.Insert(itemsIndex, item);
+						items.Insert(index, item);
 						if (externalList != null)
-							externalList.Insert(itemsIndex, item);
+							externalList.Insert(index, item);
```

Comment:
Hm, I don't think this will work after a rebuild of the filtered list - it will put the items in a different order.

Say you have 100 items, filtered down to 10.  If you insert at index 5, the new item must come between the item at index 4 and 5 of the filtered list, which is what the original code was attempting to do by inserting after the item.  The index is only relative to the filtered collection, and has no correlation to the index of the items or externalList.